### PR TITLE
Fix TLS Routing jumphost flow

### PR DESCRIPTION
**Pull request by smallinsky**
Issue: https://github.com/gravitational/teleport/issues/11271
## What
TLS Routing broke JumpHost access in case where leaf cluster is used as JumpHost. The JumpHost address is not properly propagated to `makeProxySSHClient` function call and alway root cluster is used. 

## How
Fix JumpHost address propagation. Also added support for case where JumpHost is set to Teleport proxy web port and proxy support TLS Routing.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -2288,26 +2286,61 @@ func makeProxySSHClientWithTLSWrapper(tc *TeleportClient, sshConfig *ssh.ClientC
...
+// makeProxySSHClient creates an SSH client by following steps:
+// 1) If the current proxy supports TLS Routing and JumpHost address was not provided use TLSWrapper.
+// 2) Check JumpHost raw SSH port or Teleport proxy address.
+//    In case of proxy web address check if the proxy supports TLS Routing and connect to the proxy with TLSWrapper
+// 3) Dial sshProxyAddr with raw SSH Dialer where sshProxyAddress is proxy ssh address or JumpHost address if
+//    JumpHost address was provided.
+func makeProxySSHClient(ctx context.Context, tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
+	// Use TLS Routing dialer only if proxy support TLS Routing and JumpHost was not set.
+	if tc.Config.TLSRoutingEnabled && len(tc.JumpHosts) == 0 {
+		log.Infof("Connecting proxy=%v login=%q using TLS Routing", tc.Config.WebProxyAddr, sshConfig.User)
```
Comment (r0mant) [inline lib/client/api.go:2310]:
```suggestion
		log.Infof("Connecting to proxy=%v login=%q using TLS Routing", tc.Config.WebProxyAddr, sshConfig.User)
```

**Review by r0mant — APPROVED** [review]

**Review by r0mant — COMMENTED** [review]
@smallinsky Can you also make sure it works with `ssh -J` before merging?

File: tool/tsh/proxy_test.go [hunk 2]
```diff
@@ -166,6 +174,57 @@ func testLeafClusterSSHAccess(t *testing.T, s *suite) {
 	require.NoError(t, err)
 }
 
+func testJumpHostSSHAccess(t *testing.T, s *suite) {
```
Comment (jakule) [inline tool/tsh/proxy_test.go:177]: I think converting this test cases to a test table would make this test easier to understand.

File: tool/tsh/proxy_test.go [hunk 3]
```diff
@@ -166,6 +174,57 @@ func testLeafClusterSSHAccess(t *testing.T, s *suite) {
...
+	err := Run([]string{
+		"login",
+		"--insecure",
+		"--debug",
```
Comment (jakule) [inline tool/tsh/proxy_test.go:181]: Is debug flag needed here?

File: lib/client/api.go [hunk 4]
```diff
@@ -2288,26 +2286,61 @@ func makeProxySSHClientWithTLSWrapper(tc *TeleportClient, sshConfig *ssh.ClientC
...
+
+	sshProxyAddr := tc.Config.SSHProxyAddr
+
+	// Handle situation where a Jump Host was set to proxy web address and Teleport supports TLS Routing.
+	if len(tc.JumpHosts) > 0 {
+		sshProxyAddr = tc.JumpHosts[0].Addr.Addr
+		// Check if JumpHost address is a proxy web address.
+		resp, err := webclient.Find(ctx, sshProxyAddr, tc.InsecureSkipVerify, nil)
+		// If JumpHost address is a proxy web port and proxy supports TLSRouting dial proxy with TLSWrapper.
+		if err == nil && resp.Proxy.TLSRoutingEnabled {
```
Comment (jakule) [inline lib/client/api.go:2327]: What if the error is returned here? Shouldn't be returned instead of "ignored"?

**Review by jakule — APPROVED** [review]

File: tool/tsh/proxy_test.go (hunk 3 above)
Comment (smallinsky) [reply tool/tsh/proxy_test.go:181]: No it is superfluous. Will fix.

File: lib/client/api.go (hunk 4 above)
Comment (smallinsky) [reply lib/client/api.go:2327]: This is on purpose if the call err != nil it means that proxy don't support TLS Routing and flow should proceed with standard dialer.

**Comment by smallinsky** [comment]
Tested it manually both following cases works: 
- jump host as left teleport  proxy ssh port: 
```
tsh ssh -J leaf.ice-berg.dev:33023 leafnode
```
- TLS Routing enabled when leaf WebPort can be as jump host: 
```
tsh ssh -J leaf.ice-berg.dev:33080 leafnode
```

**Pull request merged**